armi/physics/neutronics/crossSectionSettings.py

```diff
@@ -454,6 +516,8 @@ def __init__(
         self.numExternalRings = numExternalRings
         self.mergeIntoClad = mergeIntoClad
         self.meshSubdivisionsPerCm = meshSubdivisionsPerCm
+        self.xsMaxAtomNumber = xsMaxAtomNumber
+        self.minDriverDensity = minDriverDensity
```

Comment:
Do these new additions get used downstream?

Reply:
Yes

Reply:
Does it makes sense to/can we move them to the downstream plugin then? Maintaining things at the FW level that only get used in downstream plugins seems iffy/icky to me (I realize that we do this a lot throughout currently, but I think we should limit the amount of new stuff that we inject into the FW that only gets used in a downstream plugin).

Reply:
It's not a good pattern, but it's what exists. These are not the first settings to be defined like this. `nuclideReactionDriver`, `externalDriver`, and `useHomogenizedBlockComposition` are only used in downstream plugins (Also, the recently added `xsExecuteExclusive` and `xsPriority`). Additionally, many of the settings that are used in the framework are only used in methods that are only called by downstream plugins (`numInternalRings`, `numExternalRings`, `mergeIntoClad`, and `meshSubdivisionsPerCm` ).

It would be a bit messy because we'd have to subclass `XSModelingOptions` and `XSSettings` in our internal lattice physics interfaces and then use factories to get the correct types of each. I haven't attempted it, but at first glance it seems like a somewhat heavy lift. I would prefer to do it in a separate PR if possible, and then we could move all of the settings that don't belong in the framework.

Reply:
Sounds good. I wonder if it would be of interest for us to start outlining what this sort of refactor would require. @john-science any strong-feelings here?